corehq/apps/users/tests/permissions.py

```diff
@@ -94,3 +74,33 @@ def test_view_reports(self):
         self.assertFalse(self.web_user.can_view_reports(self.domain))
         self.permissions = Permissions(view_reports=True)
         self.assertTrue(self.web_user.can_view_reports(self.domain))
+
+    def test_has_permission_to_view_report_all(self):
+        self.assertFalse(has_permission_to_view_report(self.web_user, self.domain, ODATA_FEED_PERMISSION))
+        self.permissions = Permissions(view_reports=True)
+        self.assertTrue(has_permission_to_view_report(self.web_user, self.domain, ODATA_FEED_PERMISSION))
+
+    def test_has_permission_to_view_report(self):
+        self.assertFalse(has_permission_to_view_report(self.web_user, self.domain, ODATA_FEED_PERMISSION))
+        self.permissions = Permissions(view_report_list=[ODATA_FEED_PERMISSION])
+        self.assertTrue(has_permission_to_view_report(self.web_user, self.domain, ODATA_FEED_PERMISSION))
+
+    @flag_enabled('RESTRICT_APP_RELEASE')
+    def test_can_manage_releases_all(self):
+        self.permissions = Permissions(manage_releases=False)    # manage_releases is True by default
+        self.assertFalse(can_manage_releases(self.web_user, self.domain, "app_id"))
+        self.permissions = Permissions()
+        self.assertTrue(can_manage_releases(self.web_user, self.domain, "app_id"))
+
+    @flag_enabled('RESTRICT_APP_RELEASE')
+    def test_can_manage_releases(self):
+        self.permissions = Permissions(manage_releases=False)  # manage_releases is True by default
+        self.assertFalse(can_manage_releases(self.web_user, self.domain, "app_id"))
+        self.permissions = Permissions(manage_releases=False, manage_releases_list=["app_id"])
```

Comment:
Seems redundant and/or misleading to specify `manage_releases=False` when passing `manage_releases_list=[...]`. Would it be out of scope to change that in this PR?

Reply:
It's not quite redundant since a value of `True` indicates that permission is granted to all items (regardless of what's in the list). If we change this to ignore the boolean if there are items in the list we will also have to ensure that the list is cleared / set to null. I did debate this in the design doc (https://docs.google.com/document/d/1wVxmo-12dPIMwei2rn-6X9SL3GU8_KJLnY0VeVWlXdA/edit?disco=AAAAMTin2uQ) and decided to keep the boolean rather than solely relying on the the list since it is more explicit.

Reply:
I was referring to the Python API only, not the way it is stored in the database. I do think it makes sense to have a separate boolean column for `allow_all` in conjunction with a check constraint of `not (allow_all and allowed_items)`.

column values | check result
--- | ---
`allow_all = True; allowed_items = []` | TRUE
`allow_all = False; allowed_items = []` | TRUE
`allow_all = True; allowed_items = ["stuff"]` | FALSE (not allowed)
`allow_all = False; allowed_items = ["stuff"]` | TRUE

Reply:
b54283765ff